src/metabase/driver/sql_jdbc.clj

```diff
@@ -27,11 +27,20 @@
 ;;; |                                                  Run a Query                                                   |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(def ^:private ^:dynamic *tx-connections* {})
+
+(defn- local-conn
+  "If called with a transaction for the given database in dynamic scope, return the corresponding connection.
+  Otherwise, return a non-transactional connection."
+  [db-id]
+  (or (get *tx-connections* db-id)
+      (sql-jdbc.conn/db->pooled-connection-spec db-id)))
```

Comment:
I think we'll want to call `do-with-connection-with-options` here to have all the default setting right. Role, timezone etc

Reply:
I'm just wrapping the call that was already being used by all these methods, and changing that feels orthogonal to this PR.